File: data/EXP01/golang_no_system_var/2527/2527.output_2.go
func xorBeauty(nums []int) int {
    n := len(nums)
    beauty := 0

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            for k := 0; k < n; k++ {
                effective := (nums[i] | nums[j]) & nums[k]
                beauty ^= effective
            }
        }
    }

    return beauty
}

File: data/EXP01/python/2527/2527.output_2.py
# @lc app=leetcode id=2527 lang=python3
class Solution:
    def xorBeauty(self, nums: List[int]) -> int:
        n = len(nums)
        xor_beauty = 0
        
        for i in range(n):
            for j in range(n):
                for k in range(n):
                    effective_val = (nums[i] | nums[j]) & nums[k]
                    xor_beauty ^= effective_val
                    
        return xor_beauty